### PR TITLE
Update .net framework and packages nuget

### PR DESCRIPTION
**Pull request by lsaudon**
I update all project in .net 4.7.2 and update all packages.

### CODE REVIEW

**Comment by grochocki** [comment]
@lsaudon Thanks for your contribution! I started down the path of validating your changes, but ended up making a few myself (#206) to address some issues we've seen recently. In doing so, I definitely introduced merge conflicts in your PR. Can you take a look at resolving those and we can give this one another go?

**Comment by lsaudon** [comment]
Update to .net framework 4.8 and Update all packages nuget.

**Comment by grochocki** [comment]
Thanks! I'll take a closer look later tonight.

File: XamlStyler.Package/Resources.Designer.cs [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
 // <auto-generated>
-//     This code was generated by a tool.
-//     Runtime Version:4.0.30319.42000
+//     Ce code a été généré par un outil.
+//     Version du runtime :4.0.30319.42000
 //
-//     Changes to this file may cause incorrect behavior and will be lost if
-//     the code is regenerated.
+//     Les modifications apportées à ce fichier peuvent provoquer un comportement incorrect et seront perdues si
+//     le code est régénéré.
```
Comment (grochocki) [inline XamlStyler.Package/Resources.Designer.cs]: Can we revert these translation?

File: XamlStyler.Console/XamlStyler.Console.csproj [hunk 2]
```diff
@@ -27,6 +27,7 @@
     <IsWebBootstrapper>false</IsWebBootstrapper>
     <UseApplicationTrust>false</UseApplicationTrust>
     <BootstrapperEnabled>true</BootstrapperEnabled>
+    <TargetFrameworkProfile />
```
Comment (grochocki) [inline XamlStyler.Console/XamlStyler.Console.csproj]: Is this empty property required with the migration? Or left in by mistake?

File: XamlStyler.Package/app.config [hunk 3]
```diff
@@ -0,0 +1,79 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (grochocki) [inline XamlStyler.Package/app.config]: Out of curiosity, why add this file?

**Comment by grochocki** [comment]
I am seeing errors on deployment to both VS2017 and VS2019 that do not repro out of master.

**VS2017**
> CreateInstance failed for package [StylerPackage]Source: 'mscorlib' Description: Could not load file or assembly 'Microsoft.VisualStudio.Shell.15.0, Version=16.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified. System.IO.FileNotFoundException: Could not load file or assembly 'Microsoft.VisualStudio.Shell.15.0, Version=16.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified. File name: 'Microsoft.VisualStudio.Shell.15.0, Version=16.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'   at System.Reflection.RuntimeAssembly.GetType(RuntimeAssembly assembly, String name, Boolean throwOnError, Boolean ignoreCase, ObjectHandleOnStack type)   at System.Reflection.RuntimeAssembly.GetType(String name, Boolean throwOnError, Boolean ignoreCase)   at System.Activator.CreateInstanceFromInternal(String assemblyFile, String typeName, Boolean ignoreCase, BindingFlags bindingAttr, Binder binder, Object[] args, CultureInfo culture, Object[] activationAttributes, Evidence securityInfo)   at System.AppDomain.CreateInstanceFrom(String assemblyFile, String typeName)

**VS2019**
> SetSite failed for package [StylerPackage]Source: 'XamlStyler.Package' Description: Could not load file or assembly 'Microsoft.VisualStudio.Threading, Version=16.3.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified. System.IO.FileNotFoundException: Could not load file or assembly 'Microsoft.VisualStudio.Threading, Version=16.3.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified. File name: 'Microsoft.VisualStudio.Threading, Version=16.3.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'   at Xavalon.XamlStyler.Package.StylerPackage.InitializeAsync(CancellationToken cancellationToken, IProgress`1 progress)   at Microsoft.VisualStudio.Shell.AsyncPackage.<>c__DisplayClass19_0.<<Microsoft-VisualStudio-Shell-Interop-IAsyncLoadablePackageInitialize-Initialize>b__1>d.MoveNext()

File: XamlStyler.Console/XamlStyler.Console.csproj (hunk 2 above)
Comment (lsaudon) [reply XamlStyler.Console/XamlStyler.Console.csproj]: I think it's an addition when changing versions, but useless.

File: XamlStyler.Package/app.config (hunk 3 above)
Comment (lsaudon) [reply XamlStyler.Package/app.config]: I think it's a remnant before the migrate all projects from packages.config to PackageReference

File: XamlStyler.Package/Resources.Designer.cs (hunk 1 above)
Comment (lsaudon) [reply XamlStyler.Package/Resources.Designer.cs]: Sorry about my French. :wink:

File: XamlStyler.Package/app.config (hunk 3 above)
Comment (grochocki) [reply XamlStyler.Package/app.config]: Gotcha--let's remove it if it isn't important. Same with that extra property that got added to the csproj.

**Comment by grochocki** [comment]
Thanks for making those changes! I am still running into those same errors on deployment. Can you deploy on your machine?

I was able to fix it by reverting these package to the last release from 15.*:
* Microsoft.VisualStudio.Shell.15.0 - 15.9.28307
* Microsoft.VisualStudio.Threading - 15.8.209

madskristensen/AutoLoadPackageManager#2 seemed to have the exact same error with an earlier version of Threading, though the fix is undocumented, so unclear _why_ reverting fixes this. It would be great to get to the bottom of this to prevent breaks in the future.

**Comment by lsaudon** [comment]
I don't know if I can throw on my machine.
I only checked that the tests were passing 😞.
I will test tonight and change the versions of the packages.

**Comment by lsaudon** [comment]
I was able to install the extension on visual studio 2019 16.3.1

**Comment by grochocki** [comment]
Thanks, this seemed to clear up the initialization errors I was seeing. Works on VS2019, but now I am seeing this error on format:

image

No interesting logging, but will play around with it a bit more to see if I can debug.

**Comment by lsaudon** [comment]
I'm looking for ways to test integration.
CodeMaid has integration tests.
I'm not good enough to implement it for XamlStyler.

**Pull request closed without merging**